### PR TITLE
Ddd tracing

### PR DESCRIPTION
**Pull request by hrizvi1**
https://github.ibm.com/cp4i/icp4i-fermi/issues/114

### CODE REVIEW

File: products/bash/register-tracing.sh [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #
 # PARAMETERS:
 #   -n : <namespace> (string), Defaults to "cp4i"
+#   -e : <test_environment> (flag), Defaults to "false"
```
Comment (Kinshuk1993) [inline products/bash/register-tracing.sh:17]: Does this script get called somewhere with this parameter?

**Review by Kinshuk1993 — COMMENTED** [review]
Needed from within install.sh in the case repo?

File: products/bash/register-tracing.sh (hunk 1 above)
Comment (hrizvi1) [reply products/bash/register-tracing.sh:17]: yes this called from install.sh but that is to register product in main-namespace. So what is the problem here then?

**Comment by hrizvi1** [comment]
> Needed from within install.sh in the case repo?

this flag is not needed there...b/c its a flag if not provided it will assume its a main-namespace and proceed with that

File: products/bash/register-tracing.sh (hunk 1 above)
Comment (James-Kirk) [reply products/bash/register-tracing.sh:17]: I think it's only called further down during the release-mq script, it passes in -e parameter there as part of running for a 'test' namespace.

File: products/bash/release-mq.sh [hunk 2]
```diff
@@ -46,13 +50,18 @@ while getopts "n:r:i:q:t" opt; do
...
+    t ) tracing_namespace="$OPTARG"
       ;;
     \? ) usage; exit
       ;;
   esac
 done
 
+
+# tracing false if tracing namespace is empty
+[ -z $tracing_namespace ] && tracing_enabled="false"
```
Comment (James-Kirk) [inline products/bash/release-mq.sh:62]: is tracing_enabled ever actually false here?  I can't see how you'd run this script without tracing on.

File: products/bash/release-ace-integration-server.sh [hunk 3]
```diff
@@ -56,6 +63,10 @@ if [[ -z "$imageTag" ]]; then
   exit 1
 fi
 
+# tracing false if tracing namespace is empty
+[ -z $tracing_namespace ] && tracing_enabled="false"
```
Comment (James-Kirk) [inline products/bash/release-ace-integration-server.sh:67]: is tracing_enabled ever actually false here? I can't see how you'd run this script without tracing on.

**Review by James-Kirk — COMMENTED** [review]
Is there an install.sh to match these changes? Especially the release scripts?

File: DrivewayDentDeletion/Operators/cicd-dev/cicd-tasks.yaml [hunk 4]
```diff
@@ -76,4 +76,4 @@ spec:
   steps:
     - name: oc
       image: quay.io/openshift/origin-cli:latest
-      script: "/workspace/git-source/products/bash/$(params.releaseScript) -n {{NAMESPACE}} -r $(params.releaseName) -i image-registry.openshift-image-registry.svc:5000/{{NAMESPACE}}/$(params.imageName):$(params.imageTag)"
+      script: "/workspace/git-source/products/bash/$(params.releaseScript) -n {{NAMESPACE}} -r $(params.releaseName) -i image-registry.openshift-image-registry.svc:5000/{{NAMESPACE}}/$(params.imageName):$(params.imageTag) -t {{NAMESPACE}}"
```
[the remark this answers is not in the document]
Comment (hrizvi1) [reply DrivewayDentDeletion/Operators/cicd-dev/cicd-tasks.yaml]: We need to check where this task is called for in eei?
This will work for MQ and ACE-IS

**Review by DanRoseus — CHANGES_REQUESTED** [review]

File: products/bash/release-ace-integration-server.sh [hunk 5]
```diff
@@ -13,33 +13,46 @@
...
+#     ./release-ace-integration-server.sh
 #
 #   Overriding the namespace and release-name
-#     ./release-ace-is -n cp4i -r cp4i-bernie-ace
+#     ./release-ace-integration-server -n cp4i -r cp4i-bernie-ace
 
-function usage {
-    echo "Usage: $0 -n <namespace> -r <is_release_name> -i <is_image_name>"
-    exit 1
+function usage() {
```
Comment (James-Kirk) [inline products/bash/release-ace-integration-server.sh:25]: Any reason for the addition of the '()' here?

File: products/bash/release-mq.sh [hunk 6]
```diff
@@ -17,7 +17,7 @@
 #   -r : <release_name> (string), Defaults to "mq-demo"
 #   -i : <image_name> (string)
 #   -q : <qm_name> (string), Defaults to "QUICKSTART"
-#   -t : optional flag to enable tracing
```
Comment (James-Kirk) [inline products/bash/release-mq.sh:20]: How come we've removed the docs for -t, but left it in as a param?

**Comment by hrizvi1** [comment]
> Is there an install.sh to match these changes? Especially the release scripts?

yep taken care of that

File: CommonPipelineResources/cicd-tasks.yaml [hunk 7]
```diff
@@ -71,9 +71,12 @@ spec:
...
+    - name: namespace
+      type: string
+      description: release namespace
   workspaces:
     - name: git-source
   steps:
     - name: oc
       image: quay.io/openshift/origin-cli:latest
-      script: "/workspace/git-source/products/bash/$(params.releaseScript) -n {{NAMESPACE}} -r $(params.releaseName) -i image-registry.openshift-image-registry.svc:5000/{{NAMESPACE}}/$(params.imageName):$(params.imageTag)"
+      script: "/workspace/git-source/products/bash/$(params.releaseScript) -n $(params.namespace) -r $(params.releaseName) -i image-registry.openshift-image-registry.svc:5000/{{NAMESPACE}}/$(params.imageName):$(params.imageTag) -z {{NAMESPACE}}"
```
[the remark this answers is not in the document]
Comment (hrizvi1) [reply CommonPipelineResources/cicd-tasks.yaml:82]: -z is for tracing namespace and that can be different to params.namespace. Namespace will always be the one where tracing is installed. As we are installing it the main namespace where as the params.namespace can be cp4i-ddd-test and tracing is not installed in that.

File: products/bash/release-ace-integration-server.sh [hunk 8]
```diff
@@ -13,33 +13,46 @@
 #   -n : <namespace> (string), Defaults to "cp4i"
 #   -r : <is_release_name> (string), Defaults to "ace-is"
 #   -i : <is_image_name> (string), Defaults to "image-registry.openshift-image-registry.svc:5000/cp4i/ace-11.0.0.9-r2:new-1"
+#   -z : <tracing_namespace> (string), Defaults to "cp4i"
```
[the remark this answers is not in the document]
Comment (hrizvi1) [reply products/bash/release-ace-integration-server.sh:16]: fixed

File: products/bash/release-ace-integration-server.sh (hunk 5 above)
Comment (hrizvi1) [reply products/bash/release-ace-integration-server.sh:25]: fixed

File: products/bash/release-mq.sh (hunk 6 above)
Comment (hrizvi1) [reply products/bash/release-mq.sh:20]: missed it. added it now

File: DrivewayDentDeletion/Operators/cicd-test-apic/cicd-tasks.yaml [hunk 9]
```diff
@@ -79,7 +79,7 @@ spec:
   steps:
     - name: oc
       image: quay.io/openshift/origin-cli:latest
-      script: "/workspace/git-source/products/bash/$(params.releaseScript) -n $(params.namespace) -r $(params.releaseName) -i image-registry.openshift-image-registry.svc:5000/$(params.namespace)/$(params.imageName):$(params.imageTag)"
+      script: "/workspace/git-source/products/bash/$(params.releaseScript) -n $(params.namespace) -r $(params.releaseName) -i image-registry.openshift-image-registry.svc:5000/$(params.namespace)/$(params.imageName):$(params.imageTag) -z {{NAMESPACE}}"
```
[the remark this answers is not in the document]
Comment (hrizvi1) [reply DrivewayDentDeletion/Operators/cicd-test-apic/cicd-tasks.yaml:82]: -z is for tracing namespace and that can be different to params.namespace. Namespace will always be the one where tracing is installed. As we are installing it the main namespace where as the params.namespace can be cp4i-ddd-test and tracing is not installed in that.

File: products/bash/register-tracing.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (hrizvi1) [reply products/bash/register-tracing.sh:17]: Fixed it. ty

File: CommonPipelineResources/cicd-tasks.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (hrizvi1) [reply CommonPipelineResources/cicd-tasks.yaml]: fixed

**Comment by hrizvi1** [comment]
image

**Comment by hrizvi1** [comment]
image

File: products/bash/register-tracing.sh (hunk 1 above)
Comment (DanRoseus) [inline products/bash/register-tracing.sh:17]: This comment should doc the -a parameter

File: products/bash/release-ace-integration-server.sh [hunk 10]
```diff
@@ -13,36 +13,59 @@
 #   -n : <namespace> (string), Defaults to "cp4i"
 #   -r : <is_release_name> (string), Defaults to "ace-is"
 #   -i : <is_image_name> (string), Defaults to "image-registry.openshift-image-registry.svc:5000/cp4i/ace-11.0.0.9-r2:new-1"
+#   -z : <tracing_namespace> (string), Defaults to ""
```
Comment (DanRoseus) [inline products/bash/release-ace-integration-server.sh:16]: Need to doc the -t option, and what "" means for -z (I.e. same namespace as -n)

File: products/bash/release-ace-integration-server.sh [hunk 11]
```diff
@@ -85,8 +111,45 @@ spec:
...
+        echo "ERROR: Failed to register tracing in project '$namespace'"
+        exit 1
+      fi
+    else
+      if ! ${CURRENT_DIR}/register-tracing.sh -n $tracing_namespace -a ${namespace} ; then
+        echo "INFO: Running with test environment flag"
+        echo "ERROR: Failed to register tracing in project '$namespace'"
+        exit 1
+      fi
+    fi
```
Comment (DanRoseus) [inline products/bash/release-ace-integration-server.sh:148]: Can call register-tracing with -a option even if tracing_namespace == namespace, so can simplify to:
```suggestion
    if ! ${CURRENT_DIR}/register-tracing.sh -n $tracing_namespace -a ${namespace} ; then
      echo "INFO: Running with test environment flag"
      echo "ERROR: Failed to register tracing in project '$namespace'"
      exit 1
    fi
```

File: products/bash/release-mq.sh [hunk 12]
```diff
@@ -17,6 +17,7 @@
 #   -r : <release_name> (string), Defaults to "mq-demo"
 #   -i : <image_name> (string)
 #   -q : <qm_name> (string), Defaults to "QUICKSTART"
+#   -z : <tracing_namespace> (string), Defaults to ""
```
Comment (DanRoseus) [inline products/bash/release-mq.sh:20]: Doc that "" means use same namespace as -n

File: products/bash/release-mq.sh [hunk 13]
```diff
@@ -135,8 +154,36 @@ spec:
...
+          echo "ERROR: Failed to register tracing in project '$namespace'"
+          exit 1
+        fi
+    else
+       if ! ${CURRENT_DIR}/register-tracing.sh -n $tracing_namespace -a ${namespace} ; then
+          echo "INFO: Running with test environment flag"
+          echo "ERROR: Failed to register tracing in project '$namespace'"
+          exit 1
+        fi
+    fi
```
Comment (DanRoseus) [inline products/bash/release-mq.sh:179]:
```suggestion
    if ! ${CURRENT_DIR}/register-tracing.sh -n $tracing_namespace -a ${namespace} ; then
      echo "INFO: Running with test environment flag"
      echo "ERROR: Failed to register tracing in project '$namespace'"
      exit 1
    fi
```

**Review by DanRoseus — CHANGES_REQUESTED** [review]

File: products/bash/release-ace-integration-server.sh (hunk 10 above)
Comment (hrizvi1) [reply products/bash/release-ace-integration-server.sh:16]: done

File: products/bash/register-tracing.sh (hunk 1 above)
Comment (hrizvi1) [reply products/bash/register-tracing.sh:17]: done

File: products/bash/release-mq.sh (hunk 12 above)
Comment (hrizvi1) [reply products/bash/release-mq.sh:20]: fixed

File: products/bash/release-ace-integration-server.sh (hunk 11 above)
Comment (hrizvi1) [reply products/bash/release-ace-integration-server.sh]: doh...done

File: products/bash/release-mq.sh (hunk 13 above)
Comment (hrizvi1) [reply products/bash/release-mq.sh]: doh..done

**Review by DanRoseus — APPROVED** [review]

**Pull request merged**